iroha/src/sumeragi.rs

```diff
@@ -45,8 +51,65 @@ impl Message {
     }
 }
 
+/// Structure describing a request to other peers to change view because of the commit timeout.
+#[derive(Io, Decode, Encode, Debug, Clone)]
+pub struct CommitTimeout {
+    voting_block_hash: Hash,
+    signatures: Vec<Signature>,
+}
+
+impl CommitTimeout {
+    /// Initializes a new commit timeout message with no signatures
+    pub fn new(voting_block: VotingBlock) -> CommitTimeout {
+        CommitTimeout {
+            voting_block_hash: voting_block.block.hash(),
+            signatures: vec![],
+        }
+    }
+
+    /// Signes this request with the peer's public and private key.
+    /// This way peers vote for changing the view.
+    pub fn sign(&mut self, public_key: &PublicKey, private_key: &PrivateKey) -> Result<(), String> {
+        let already_signed = self
+            .signatures
+            .iter()
+            .any(|signature| signature.public_key() == public_key);
+        if !already_signed {
+            self.signatures.push(Signature::new(
+                *public_key,
+                &self.voting_block_hash,
+                private_key,
+            )?);
+            Ok(())
+        } else {
+            Err("Already signed.".to_string())
+        }
```

Comment:
Maybe it's better to use duplicates free structure as signatures storage? It will make it impossible to add the same signature twice.

Reply:
I just realized that parity scale codec can not encode HashSets / HashMaps, so let's stay with Vec for now. Anyway we need to know if the signature was added or not, so to decide whether to send a message.